### PR TITLE
Molgan GPU Support

### PR DESCRIPTION
**Pull request by shreyasvinaya**
## Description

add molgan gpu support

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [ ] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [ ] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [ ] Run `mypy -p deepchem` and check no errors
  - [ ] Run `flake8 <modified file> --count` and check no errors
  - [ ] Run `python -m doctest <modified file>` and check no errors
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New unit tests pass locally with my changes
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by rbharath** [comment]
@shreyasvinaya I'm seeing major test failures. Look at unit test 3.8. Can you debug?

File: deepchem/models/torch_models/gan.py [hunk 1]
```diff
@@ -716,14 +716,8 @@ def __init__(self,
         """
         self.n_generators = n_generators
         self.n_discriminators = n_discriminators
+        self.device = device  # type: ignore
```
Comment (rbharath) [inline deepchem/models/torch_models/gan.py:719]: This needs to be changed to be beginner friendly; a beginner with a GPU should have it used by default

**Review by rbharath — COMMENTED** [review]
A change requested for beginner friendliness

File: deepchem/models/torch_models/gan.py (hunk 1 above)
Comment (shreyasvinaya) [reply deepchem/models/torch_models/gan.py]: fixed

**Pull request merged**